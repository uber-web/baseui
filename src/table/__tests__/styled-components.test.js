/*
Copyright (c) 2018 Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/
// @flow

import React from 'react';
import {shallow} from 'enzyme';
import {
  StyledTable,
  StyledHead,
  StyledHeadCell,
  StyledBody,
  StyledRow,
  StyledCell,
  StyledFilterButton,
  StyledFilterContent,
  StyledFilterHeading,
  StyledFilterFooter,
  StyledAction,
} from '../index.js';

describe('Table styled components', () => {
  const styledComponents = [
    [StyledTable, 'StyledTable'],
    [StyledHead, 'StyledHead'],
    [StyledHeadCell, 'StyledHeadCell'],
    [StyledBody, 'StyledBody'],
    [StyledRow, 'StyledRow'],
    [StyledCell, 'StyledCell'],
    [StyledFilterButton, 'StyledFilterButton'],
    [StyledFilterContent, 'StyledFilterContent'],
    [StyledFilterHeading, 'StyledFilterHeading'],
    [StyledFilterFooter, 'StyledFilterFooter'],
    [StyledAction, 'StyledAction'],
  ];

  test.each(styledComponents)('default properties', (Component, name) => {
    const component = shallow(<Component />);
    expect(component.instance().getStyles()).toMatchSnapshot(
      `${name} has expected default styles`,
    );
  });

  test.each(styledComponents)('$width property', (Component, name) => {
    const component = shallow(<Component $width="1000px" />);
    expect(component.instance().getStyles()).toMatchSnapshot(
      `${name} has expected styles when $width property defined`,
    );
  });
});
