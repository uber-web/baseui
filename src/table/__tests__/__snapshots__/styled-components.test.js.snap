// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table styled components $width property: StyledAction has expected styles when $width property defined 1`] = `
Object {
  ":hover:enabled": Object {
    "cursor": "pointer",
  },
  "border": "none",
  "color": "$theme.colors.primary",
  "marginLeft": "$theme.sizing.scale100",
  "marginRight": "$theme.sizing.scale100",
  "paddingBottom": 0,
  "paddingLeft": 0,
  "paddingRight": 0,
  "paddingTop": 0,
}
`;

exports[`Table styled components $width property: StyledBody has expected styles when $width property defined 1`] = `
Object {
  "flex": 1,
  "overflowX": "hidden",
  "overflowY": "overlay",
  "width": "1000px",
}
`;

exports[`Table styled components $width property: StyledCell has expected styles when $width property defined 1`] = `
Object {
  "display": "flex",
  "flex": 1,
  "fontFamily": "$theme.typography.font300.fontFamily",
  "fontSize": "$theme.typography.font300.fontSize",
  "fontWeight": "$theme.typography.font300.fontWeight",
  "lineHeight": "$theme.typography.font300.lineHeight",
  "paddingBottom": "$theme.sizing.scale300",
  "paddingLeft": "$theme.sizing.scale600",
  "paddingRight": "$theme.sizing.scale600",
  "paddingTop": "$theme.sizing.scale300",
}
`;

exports[`Table styled components $width property: StyledFilterButton has expected styles when $width property defined 1`] = `
Object {
  "border": "none",
  "padding": "none",
}
`;

exports[`Table styled components $width property: StyledFilterContent has expected styles when $width property defined 1`] = `
Object {
  "borderColor": "$theme.borders.border300.borderColor",
  "borderLeft": "none",
  "borderRight": "none",
  "borderStyle": "$theme.borders.border300.borderStyle",
  "borderWidth": "$theme.borders.border300.borderWidth",
  "maxHeight": "196px",
  "overflow": "auto",
  "paddingLeft": "$theme.sizing.scale600",
  "paddingRight": "$theme.sizing.scale600",
}
`;

exports[`Table styled components $width property: StyledFilterFooter has expected styles when $width property defined 1`] = `
Object {
  "backgroundColor": "$theme.colors.mono200",
  "display": "flex",
  "justifyContent": "space-between",
  "paddingBottom": "$theme.sizing.scale300",
  "paddingLeft": "$theme.sizing.scale100",
  "paddingRight": "$theme.sizing.scale100",
  "paddingTop": "$theme.sizing.scale300",
  "width": "216px",
}
`;

exports[`Table styled components $width property: StyledFilterHeading has expected styles when $width property defined 1`] = `
Object {
  "color": "$theme.colors.mono700",
  "fontFamily": "$theme.typography.font350.fontFamily",
  "fontSize": "$theme.typography.font350.fontSize",
  "fontWeight": "$theme.typography.font350.fontWeight",
  "lineHeight": "$theme.typography.font350.lineHeight",
  "paddingBottom": "$theme.sizing.scale500",
  "paddingLeft": "$theme.sizing.scale600",
  "paddingRight": "$theme.sizing.scale600",
  "paddingTop": "$theme.sizing.scale500",
}
`;

exports[`Table styled components $width property: StyledHead has expected styles when $width property defined 1`] = `
Object {
  "backgroundColor": "$theme.colors.tableHeadBackgroundColor",
  "boxShadow": "$theme.lighting.shadow400",
  "display": "flex",
  "flexGrow": 0,
  "width": "1000px",
  "zIndex": 1,
}
`;

exports[`Table styled components $width property: StyledHeadCell has expected styles when $width property defined 1`] = `
Object {
  ":last-of-type": Object {
    "borderRight": "none",
  },
  "borderBottom": "none",
  "borderColor": "$theme.borders.border300.borderColor",
  "borderLeft": "none",
  "borderStyle": "$theme.borders.border300.borderStyle",
  "borderTop": "none",
  "borderWidth": "$theme.borders.border300.borderWidth",
  "display": "flex",
  "fontFamily": "$theme.typography.font350.fontFamily",
  "fontSize": "$theme.typography.font350.fontSize",
  "fontWeight": "$theme.typography.font350.fontWeight",
  "justifyContent": "space-between",
  "lineHeight": "$theme.typography.font350.lineHeight",
  "paddingBottom": "$theme.sizing.scale500",
  "paddingLeft": "$theme.sizing.scale600",
  "paddingRight": "$theme.sizing.scale600",
  "paddingTop": "$theme.sizing.scale500",
  "width": "100%",
}
`;

exports[`Table styled components $width property: StyledRow has expected styles when $width property defined 1`] = `
Object {
  "alignItems": "center",
  "display": "flex",
}
`;

exports[`Table styled components $width property: StyledTable has expected styles when $width property defined 1`] = `
Object {
  "borderColor": "$theme.borders.border300.borderColor",
  "borderRadius": "$theme.borders.radius200",
  "borderStyle": "$theme.borders.border300.borderStyle",
  "borderWidth": "$theme.borders.border300.borderWidth",
  "display": "flex",
  "flexDirection": "column",
  "height": "100%",
  "overflowX": "scroll",
}
`;

exports[`Table styled components default properties: StyledAction has expected default styles 1`] = `
Object {
  ":hover:enabled": Object {
    "cursor": "pointer",
  },
  "border": "none",
  "color": "$theme.colors.primary",
  "marginLeft": "$theme.sizing.scale100",
  "marginRight": "$theme.sizing.scale100",
  "paddingBottom": 0,
  "paddingLeft": 0,
  "paddingRight": 0,
  "paddingTop": 0,
}
`;

exports[`Table styled components default properties: StyledBody has expected default styles 1`] = `
Object {
  "flex": 1,
  "overflowX": "hidden",
  "overflowY": "overlay",
  "width": "100%",
}
`;

exports[`Table styled components default properties: StyledCell has expected default styles 1`] = `
Object {
  "display": "flex",
  "flex": 1,
  "fontFamily": "$theme.typography.font300.fontFamily",
  "fontSize": "$theme.typography.font300.fontSize",
  "fontWeight": "$theme.typography.font300.fontWeight",
  "lineHeight": "$theme.typography.font300.lineHeight",
  "paddingBottom": "$theme.sizing.scale300",
  "paddingLeft": "$theme.sizing.scale600",
  "paddingRight": "$theme.sizing.scale600",
  "paddingTop": "$theme.sizing.scale300",
}
`;

exports[`Table styled components default properties: StyledFilterButton has expected default styles 1`] = `
Object {
  "border": "none",
  "padding": "none",
}
`;

exports[`Table styled components default properties: StyledFilterContent has expected default styles 1`] = `
Object {
  "borderColor": "$theme.borders.border300.borderColor",
  "borderLeft": "none",
  "borderRight": "none",
  "borderStyle": "$theme.borders.border300.borderStyle",
  "borderWidth": "$theme.borders.border300.borderWidth",
  "maxHeight": "196px",
  "overflow": "auto",
  "paddingLeft": "$theme.sizing.scale600",
  "paddingRight": "$theme.sizing.scale600",
}
`;

exports[`Table styled components default properties: StyledFilterFooter has expected default styles 1`] = `
Object {
  "backgroundColor": "$theme.colors.mono200",
  "display": "flex",
  "justifyContent": "space-between",
  "paddingBottom": "$theme.sizing.scale300",
  "paddingLeft": "$theme.sizing.scale100",
  "paddingRight": "$theme.sizing.scale100",
  "paddingTop": "$theme.sizing.scale300",
  "width": "216px",
}
`;

exports[`Table styled components default properties: StyledFilterHeading has expected default styles 1`] = `
Object {
  "color": "$theme.colors.mono700",
  "fontFamily": "$theme.typography.font350.fontFamily",
  "fontSize": "$theme.typography.font350.fontSize",
  "fontWeight": "$theme.typography.font350.fontWeight",
  "lineHeight": "$theme.typography.font350.lineHeight",
  "paddingBottom": "$theme.sizing.scale500",
  "paddingLeft": "$theme.sizing.scale600",
  "paddingRight": "$theme.sizing.scale600",
  "paddingTop": "$theme.sizing.scale500",
}
`;

exports[`Table styled components default properties: StyledHead has expected default styles 1`] = `
Object {
  "backgroundColor": "$theme.colors.tableHeadBackgroundColor",
  "boxShadow": "$theme.lighting.shadow400",
  "display": "flex",
  "flexGrow": 0,
  "width": "100%",
  "zIndex": 1,
}
`;

exports[`Table styled components default properties: StyledHeadCell has expected default styles 1`] = `
Object {
  ":last-of-type": Object {
    "borderRight": "none",
  },
  "borderBottom": "none",
  "borderColor": "$theme.borders.border300.borderColor",
  "borderLeft": "none",
  "borderStyle": "$theme.borders.border300.borderStyle",
  "borderTop": "none",
  "borderWidth": "$theme.borders.border300.borderWidth",
  "display": "flex",
  "fontFamily": "$theme.typography.font350.fontFamily",
  "fontSize": "$theme.typography.font350.fontSize",
  "fontWeight": "$theme.typography.font350.fontWeight",
  "justifyContent": "space-between",
  "lineHeight": "$theme.typography.font350.lineHeight",
  "paddingBottom": "$theme.sizing.scale500",
  "paddingLeft": "$theme.sizing.scale600",
  "paddingRight": "$theme.sizing.scale600",
  "paddingTop": "$theme.sizing.scale500",
  "width": "100%",
}
`;

exports[`Table styled components default properties: StyledRow has expected default styles 1`] = `
Object {
  "alignItems": "center",
  "display": "flex",
}
`;

exports[`Table styled components default properties: StyledTable has expected default styles 1`] = `
Object {
  "borderColor": "$theme.borders.border300.borderColor",
  "borderRadius": "$theme.borders.radius200",
  "borderStyle": "$theme.borders.border300.borderStyle",
  "borderWidth": "$theme.borders.border300.borderWidth",
  "display": "flex",
  "flexDirection": "column",
  "height": "100%",
  "overflowX": "scroll",
}
`;
